src/db.c

```diff
@@ -126,7 +126,7 @@ robj *lookupKey(serverDb *db, robj *key, int flags) {
          * Don't do it if we have a saving child, as this will trigger
          * a copy on write madness. */
         if (server.current_client && server.current_client->flag.no_touch &&
-            server.executing_client->cmd->proc != touchCommand)
+            server.executing_client && server.executing_client->cmd->proc != touchCommand)
```

Comment:
In #1499, I originally proposed to fall back to the `current_client` if the `executing_client` was NULL. I wonder if that's better as the `current_client` was used in this check before #1499.

That being said, I think it is odd that the `executing_client` is not set when we are re-processing an unblocked blocking command. @ranshid Do you know whether this is intentional?

Reply:
'executing_client' is valid inside of 'call' only. The unblocking of a blocked client occurs after the running client finishes its command (that triggers the unblocking). This particular lookup is actually done with "NOTOUCH".

Reply:
Yes. the executing_client was an evolution of the different client types we once had like script_caller. before that we had many cases were we had to change the current_client many times during a flow of operations. BTW, It is valid even outside of "call' flows in cases were we execute commands outside of processing context (eg AOF load)